pytrip/lib/core.c

```diff
@@ -751,6 +751,61 @@ static PyObject * slice_on_plane(PyObject *self, PyObject *args)
         }
     }
 
+    // choose intersection plane type
```

Comment:
Can't this be compressed into single loop with one more pass and some extra `if` statement, like:

```
if( i < length) {
 first_index = i;
 second_index = i+1;
} else {
 first_index = length;
 second_index = 0;
} 

Also - add a comment mentioning that this is needed to close the loop.

Reply:
I think it can worsen efficiency, at the end of the for loop we know that the loop has ended, we don't need to check that in every step of loop as in Your suggestion

Reply:
Yes, but it improved code readability, as now we have huge DYI violation by large block being copy-pasted

Reply:
fast code >>> good looking code :)
but i will test that suggestions

Reply:
did you mean DRY? Don't repeat yourself?

Reply:
yes

you can also calculate index with modulo division:

```
first_index = i % length
second_index = (i+1) % length
```